pallets/parachain-staking/src/lib.rs

```diff
@@ -2284,6 +2324,20 @@ pub mod pallet {
 			// prepare unstaking of collator candidate
 			Self::prep_unstake_exit_queue(&state.id, state.stake);
 
+			// disable validator for next session if they were in the set of validators
+			if let Some(index) = pallet_session::Pallet::<T>::validators()
+				.into_iter()
+				.enumerate()
+				.find_map(|(i, id): (usize, T::ValidatorId)| {
```

Comment:
Why do we need types here?

```suggestion
				.find_map(|(i, id)| {
```

Reply:
Fixed in e547d10